go/test/stress/stress.go

```diff
@@ -0,0 +1,404 @@
+// The method will then start a goroutine that will spawn one or more clients.
+// These clients will be responsible for stressing the cluster until Config.MaximumDuration
+// is reached, or until Stresser.Stop() or Stresser.StopAfter() are called.
+//
+// This method returns a pointer to its Stresser to allow chained function call, like:
+//
+// 		s := stress.New(t, cfg).Start()
+//		s.Stop()
+//
+func (s *Stresser) Start() *Stresser {
+	fmt.Println("Starting load testing ...")
+	s.tbls = s.createTables(s.cfg.NumberOfTables)
+	s.start = time.Now()
+	go s.startClients()
+	return s
+}
+
+func generateNewTables(prefix string, nb int) []*table {
+	tbls := make([]*table, 0, nb)
+	for i := 0; i < nb; i++ {
+		tbls = append(tbls, &table{
+			name: fmt.Sprintf("%s%d", prefix, i),
+		})
+	}
+	return tbls
+}
+
+func (s *Stresser) createTables(nb int) []*table {
+	conn := newClient(s.t, s.cfg.ConnParams)
+	defer conn.Close()
+
+	tbls := generateNewTables(s.cfg.TableNamePrefix, nb)
+	for _, tbl := range tbls {
+		s.exec(conn, fmt.Sprintf(templateNewTable, tbl.name))
+	}
+	return tbls
+}
+
+// startClients is responsible for concurrently starting all the clients,
+// fetching their results, and computing a single final result which is
+// then publish in Stresser.doneCh.
+func (s *Stresser) startClients() {
+	maxClient := s.cfg.MaxClient
+	resultCh := make(chan result, maxClient)
+
+	// Start the concurrent clients.
+	for i := 0; i < maxClient; i++ {
+		go s.startStressClient(resultCh)
+	}
+
+	// Wait for the different clients to publish their results.
+	perClientResults := make([]result, 0, maxClient)
+	for i := 0; i < maxClient; i++ {
+		newResult := <-resultCh
+		perClientResults = append(perClientResults, newResult)
+	}
+
+	// Calculate how long it took for all the client to finish stressing
+	// the cluster.
+	s.duration = time.Since(s.start)
+
+	// Based on all the clients' results, compute a single result.
+	var finalResult result
+	for _, r := range perClientResults {
+		finalResult.inserts = sumQueryCounts(finalResult.inserts, r.inserts)
+		finalResult.selects = sumQueryCounts(finalResult.selects, r.selects)
+		finalResult.deletes = sumQueryCounts(finalResult.deletes, r.deletes)
+	}
+	s.doneCh <- finalResult
+}
+
+// startStressClient creates a client that will stress the cluster.
+// This function is supposed to be called as many times as we want
+// to have concurrent clients stressing the cluster.
+// Once the client is done stressing the cluster, results are published
+// in the given chan result.
+func (s *Stresser) startStressClient(resultCh chan result) {
+	s.cfgMu.Lock()
+	connParams := s.cfg.ConnParams
+	s.cfgMu.Unlock()
+
+	conn := newClient(s.t, connParams)
+	defer conn.Close()
+
+	var res result
+
+	// Create a timeout based on the Stresser maximum duration and the time
+	// that has already elapsed since the Stresser was started.
+	timeout := time.After(s.cfg.MaximumDuration - time.Since(s.start))
+
+outer:
+	for !s.finish {
```

Comment:
This is not atomic, and the writes to `s.finish` from `StopAfter` are not atomic either. For sanity's sake, I would recommend making `s.finish` an `uint32` and using `atomic.{Load,Store}Uint32` to perform this check.